updated peacock ability, fixed #46 and fixed hatching chick lvl 3

jimkaal opened the pull request:
Updated peacock to 2/5
Modified ability to be 50% of attack with max triggers

Addresses part of #32

alexdriedger commented on tests/test_effects.py:
```diff
@@ -180,3 +180,10 @@ def test_weak(self):
         attack_phase = get_attack(fish, t3[0].pet)
         self.assertEqual(attack_phase[1], 9) # 6/8 + 3
+
+    def test_peacock(self):
```
It would be good if the test made sure that it triggers the correct number of times

jimkaal commented:
Fixed #46 
in battles, since butterfly always activates TransferStats with attack 1 priority in StartOfBattle, i added a check after all StartOfBattle triggers that checks for self summons (currently only butterfly). This makes sure butterfly summon has the correct priority compared to snipers.
in effects - evolve, made sure to create a new pet instead of just the name and removed the new pet effect trigger, since the above priority issue

added unittests for both high and low attack caterpillars to test the pet priority

jimkaal commented:
Thank you for that comment, I put the "maxTriggers" in the wrong place
Modified the unittest and fixed the issue

jimkaal commented:
Apparently hatching chick level 3 was not added to the StartOfTurn triggers nor unittests.
Immediately threw an error when its ability was used, GainExperience used target pet ability to determine experience amount, which only works for caterpillar. Modified this to now check the triggering pet's ability.
